src/lib/fcitx-utils/i18n.cpp

```diff
@@ -22,8 +22,11 @@ class GettextManager {
         if (domains_.count(domain)) {
             return;
         }
-        const auto *localedir = StandardPath::fcitxPath("localedir");
         if (!dir) {
+            const auto *localedir = getenv("FCITX_LOCALE_DIR");
```

Comment:
I suggest you make FCITX_LOCALE_DIR to overrides everything regardless what is being passed to registerDomain.

Reply:
Would it affet fcitx5 itself? I can't find call to `registerDomain("fcitx5", ...)` in the repo ...
But if I call `registerDomain("fcitx5", ...)` manually before it's ready, translations will work as expected.